aten/src/ATen/cpu/vec256/vec256_float_neon.h

```diff
@@ -642,6 +642,853 @@ Vec256<float> inline fmadd(const Vec256<float>& a, const Vec256<float>& b, const
+        }
+      case 7:
+        {
+          Vec256<float> vec;
+          static uint32x4_t mask_high = {0xFFFFFFFF, 0xFFFFFFFF, 0xFFFFFFFF, 0x0};
+          vec.values.val[0] = b.values.val[0];
+          vec.values.val[1] = (float32x4_t)mask_high;
+          __asm__ (
+              "vbsl.f32 %[res_mask], %[in_b], %[in_a]\n\t"
+              : [res_mask] "+w" (vec.values.val[1])
+              : [in_b] "w" (b.values.val[1]), [in_a] "w" (a.values.val[1])
+              :);
+          return vec;
+        }
+    }
+    return b;
+  }
+  static Vec256<float> loadu(const void* ptr, int64_t count = size()) {
+    if (count == size()) {
+      volatile register float32x4_t low asm("q3");
+      volatile register float32x4_t high asm("q4");
+      __asm__ __volatile__ (
+          "vld1.32 {d6, d7, d8, d9}, [%[in_ptr]]\n\t"
+          : "=w" (low), "=w" (high)
+          : [in_ptr] "r" (ptr)
+          :);
+      return Vec256<float>(low, high);
+    }
+    else if (count == (size() >> 1)) {
+      volatile register float32x4_t low asm("q3");
+      float32x4_t high{0.f, 0.f, 0.f, 0.f};
+      Vec256<float> res;
+      __asm__  __volatile__ (
+          "vld1.32 {d6, d7}, [%[in_ptr]]\n\t"
+          : "=w" (low)
+          : [in_ptr] "r" (ptr)
+          :);
+      return Vec256<float>(low, high);
+    }
+    else {
+      __at_align32__ float tmp_values[size()];
+      for (auto i = 0; i < size(); ++i) {
+        tmp_values[i] = 0.0;
+      }
+      std::memcpy(
+          tmp_values,
+          reinterpret_cast<const float*>(ptr),
+          count * sizeof(float));
+      volatile register float32x4_t low asm("q3");
+      volatile register float32x4_t high asm("q4");
+      __asm__  __volatile__ (
+          "vld1.32 {d6, d7, d8, d9}, [%[in_ptr]]\n\t"
+          : "=w" (low), "=w" (high)
+          : [in_ptr] "r" (tmp_values)
+          :);
+      return Vec256<float>(low, high);
+    }
+    return Vec256<float>();
+  }
+  void store(void* ptr, int64_t count = size()) const {
+    if (count == size()) {
+      __asm__  __volatile__ (
+          "vmov.f32 q3, %[in_low]\n\t"
+          "vmov.f32 q4, %[in_high]\n\t"
+          "vst1.32 {d6, d7, d8, d9}, [%[in_ptr]]\n\t"
+          :
+          : [in_ptr] "r" (ptr),
+            [in_low] "w" (values.val[0]), [in_high] "w" (values.val[1])
+          : "q3", "q4");
+    }
+    else if (count == (size() >> 1)) {
+      __asm__  __volatile__ (
+          "vmov.f32 q3, %[in_low]\n\t"
+          "vst1.32 {d6, d7}, [%[in_ptr]]\n\t"
+          :
+          : [in_ptr] "r" (ptr), [in_low] "w" (values.val[0])
+          : "q3");
+    }
+    else {
+      float tmp_values[size()];
+      __asm__  __volatile__ (
+          "vmov.f32 q3, %[in_low]\n\t"
+          "vmov.f32 q4, %[in_high]\n\t"
+          "vst1.32 {d6, d7, d8, d9}, [%[in_ptr]]\n\t"
+          :
+          : [in_ptr] "r" (tmp_values),
+            [in_low] "w" (values.val[0]), [in_high] "w" (values.val[1])
+          : "q3", "q4");
```

Comment:
Again "memory" here and everywhere else you are doing a read or write memory access."

Reply:
I think it makes ordering requirement stringent and prevents compiler from doing any optimization, as that makes it look like a barrier, I think. Hence I did not do it.

Reply:
I am not sure it is optional.  I was under the impression that it is required for correct behavior.

Reply:
@AshkanAliabadi, so this one seems to suggest that it prevents compiler from reordering loads/stores before "asm". https://stackoverflow.com/questions/14449141/the-difference-between-asm-asm-volatile-and-clobbering-memory. Basically it makes it assume that you dont know what memory gets written inside "asm" so dont reorder load/stores. But since address registers are linked in the asm I assume compiler understands what gets written? I was trying not to put constraint on compiler optimization.
Seems like modifying this to "rm" should be ok: https://gcc.gnu.org/onlinedocs/gcc/Extended-Asm.html

Reply:
@AshkanAliabadi, doesnt seem that I can get "rm" to work so just added "memory" to the clobbers list as you suggested.